frontend/amundsen_application/static/js/pages/HomePage/index.tsx

```diff
@@ -47,28 +91,7 @@ export class HomePage extends React.Component<HomePageProps> {
             }`}
           >
             <h1 className="sr-only">{HOMEPAGE_TITLE}</h1>
-            <div className="home-element-container">
-              <SearchBar />
-            </div>
-            <div className="filter-breadcrumb pull-right">
-              <Breadcrumb
-                direction="right"
-                path="/search"
-                text={SEARCH_BREADCRUMB_TEXT}
-              />
-            </div>
-            <div className="home-element-container">
-              <BadgesListContainer shortBadgesList />
-            </div>
-            <div className="home-element-container">
-              <TagsListContainer shortTagsList />
-            </div>
-            <div className="home-element-container">
-              <MyBookmarks />
-            </div>
-            <div className="home-element-container">
-              <PopularTables />
-            </div>
+            {this.homePageWidgets}
```

Comment:
What do you mean by having new components rendered? A new HomePageWidgetContainer (or whatever name) component that holds all the widgets that the user wants on their homepage? That seems logical to me.

Reply:
So it could be something like:

```
  render() {
    /* TODO, just display either popular or curated tags,
    do we want the title to change based on which
    implementation is being used? probably not */
    return (
      <main className="container home-page">
        <div className="row">
          <div
            className={`col-xs-12 ${
              announcementsEnabled() ? 'col-md-8' : 'col-md-offset-1 col-md-10'
            }`}
          >
            <h1 className="sr-only">{HOMEPAGE_TITLE}</h1>
            ****<HomePageWidgets
               getHomePageWidgets={getHomePageWidgets} 
             />****
          </div>
          {announcementsEnabled() && (
            <div className="col-xs-12 col-md-offset-1 col-md-3">
              <Announcements />
            </div>
          )}
        </div>
      </main>
    );
  }
}
```

Reply:
Yes, but you don't need to extract it into a different file for the moment, you can write it above line 40 as a functional component.

Reply:
Thanks, will do!